sql/catalyst/src/main/scala/org/apache/spark/sql/QueryExecutionErrors.scala

```diff
@@ -51,18 +58,209 @@ object QueryExecutionErrors {
       s"[BUG] logical plan should not have output of char/varchar type: $leaf")
   }
 
-  def cannotEvaluateGeneratorError(generator: UnresolvedGenerator): Throwable = {
-    new UnsupportedOperationException(s"Cannot evaluate expression: $generator")
+  def cannotEvaluateExpressionError(expression: Expression): Throwable = {
+    new UnsupportedOperationException(s"Cannot evaluate expression: $expression")
   }
 
-  def cannotGenerateCodeForGeneratorError(generator: UnresolvedGenerator): Throwable = {
-    new UnsupportedOperationException(s"Cannot generate code for expression: $generator")
+  def cannotGenerateCodeForExpressionError(expression: Expression): Throwable = {
+    new UnsupportedOperationException(s"Cannot generate code for expression: $expression")
   }
 
   def cannotTerminateGeneratorError(generator: UnresolvedGenerator): Throwable = {
     new UnsupportedOperationException(s"Cannot terminate expression: $generator")
   }
 
+  def castDownLongCauseOverflowError(t: Any, targetType: String): Throwable = {
+    new ArithmeticException(s"Casting $t to $targetType causes overflow")
+  }
+
+  def cannotRepresentDecimalError(value: Decimal, decimalType: DecimalType): Throwable = {
+    new ArithmeticException(s"${value.toDebugString} cannot be represented as " +
+      s"Decimal(${decimalType.precision}, ${decimalType.scale}).")
+  }
+
+  def invalidInputSyntaxError(s: UTF8String): Throwable = {
+    new NumberFormatException(s"invalid input syntax for type numeric: $s")
+  }
+
+  def cannotCastFromNullTypeDirectlyError(to: DataType): Throwable = {
+    new SparkException(s"should not directly cast from NullType to $to.")
+  }
+
+  def cannotCastError(from: DataType, to: DataType): Throwable = {
+    new SparkException(s"Cannot cast $from to $to.")
+  }
+
+  def cannotParseDecimalError(): Throwable = {
+    new IllegalArgumentException("Cannot parse any decimal")
+  }
+
+  def simpleStringWithNodeIdUnsupportedError(nodeName: String): Throwable = {
+    new UnsupportedOperationException(s"$nodeName does not implement simpleStringWithNodeId")
+  }
+
+  def evaluateUnevaluableAggregateUnsupportedError(
```

Comment:
unsupportedEvaluateUnevaluableAggregateError? (Basically, its better to use `unsupported` for the prefix?)

Reply:
There exist a lot of `operatorUnsupportedError`.  The operator unsupported error.

Reply:
I am neutral about using `unsupported` as a prefix or a suffix as long as the usage is consistent.